src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageParser.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class NotificationMessageParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationMessageParser.class);
+    private static final String ERROR_CAUSE = "Message Binary data is null";
+
+    private final ObjectMapper objectMapper;
+
+    public NotificationMessageParser(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public NotificationMsg parse(MessageBody messageBody) {
+        try {
+            NotificationMsg notificationMsg =
+                    objectMapper.readValue(getBinaryData(messageBody), NotificationMsg.class);
+            LOGGER.info(
+                    "Parsed notification message, Zip File Name: {}, Error Code: {}, Error Description: {} "
+                            + "Jurisdiction: {}, PO Box: {}, Service: {}, Document Control Number: {}",
+                    notificationMsg.zipFileName,
+                    notificationMsg.errorCode,
+                    notificationMsg.errorDescription,
+                    notificationMsg.jurisdiction,
+                    notificationMsg.poBox,
+                    notificationMsg.service,
+                    notificationMsg.documentControlNumber
```

Comment:
looks like custom `toString` method come in handy for `notificationMsg` in this case

Reply:
if we use tostring we create a string for every log , which I guess we do not want.

Reply:
we interpolate all those strings for every log anyway. each object has `toString` method which can be overwritten and in here just directly passed to log message interpolation

Reply:
for each log we concat all these fields and create another new string ?
and even we increase the log level from info when we pass this param to log we still continue creating new strings.

Reply:
this is stale now